.github/workflows/reusable-build.yml

```diff
@@ -181,12 +181,18 @@ jobs:
       working-directory: ${{env.GITHUB_WORKSPACE}}
       run: msbuild /m /p:Configuration=${{env.BUILD_CONFIGURATION}} /p:Platform=${{env.BUILD_PLATFORM}} ${{env.BUILD_OPTIONS}} ${{env.SOLUTION_FILE_PATH}}
 
-    - name: Check binary dependencies
+    - name: Check DLL dependencies for distributed binaries
       if: steps.skip_check.outputs.should_skip != 'true' && ((inputs.build_artifact == 'Build-x64' && matrix.configurations == 'Debug') || (inputs.build_artifact == 'Build-x64-native-only' && matrix.configurations == 'NativeOnlyRelease'))
       working-directory: ./${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
       run: |
         ..\..\scripts\check_binary_dependencies.ps1 -BuildArtifact "${{inputs.build_artifact}}_${{env.BUILD_CONFIGURATION}}" -VsToolsPath "${{env.msvc_tools_path}}"
 
+    - name: Test the MSI
```

Comment:
1. I think this should be moved out of the build step to avoid extending the critical path for the CI.
Our `driver `tests implicitly verify that eBPF drivers can be installed - otherwise they wouldn't run - so this test is specific to the MSI installer containing all the necessary files and working properly. 

2. We could also limit this test step to just checking the required files are in place, and move the installation part to the `driver_ws2019/2022` jobs to replace their use of the PS scripts for installation. Their cleanup step would exercise the uninstall path. This would also remove the (un)/installation happening twice, even if in parallel on separate machines.

Reply:
1. The MSI is not just a container, it copies all its files and then *installs* them (i.e. drivers, netsh extension and JIT service), in addition to checking preconditions in the environment (which may cause some checks to fail). This is why the tests cover such operations, otherwise not covered anywhere, specifically for an MSI-only deployment. So, the point here is not to check "if the drivers work/can be installed", but rather "if the MSI installs *everything* correctly".
Being the MSI a release deliverable, built with source code, all its operations must be tested, in order to avoid regressions that have already happened in the past.

1. Without an installation, the files would not be there to check. The installation and uninstallation are the most time-consuming operations, whereas all the tests in between are almost instant, so imho I don't see gains in doing them partially (btw, being the file checking the "slowest", eventually).

If we want to move *all this atomically* to another section of the pipeline, happy to discuss any suggestion, but imho this would be the best way.

Reply:
So I was making two points:
1. let's not increase the duration of the critical path
2. let's use this opportunity to have the driver19/22 jobs use the MSI.

Re. 1 Looking at the time this new MSI step is taking, it's actually **not** happening, so treat extracting this step as optional.
The Release build is still the long pole in the build step, even with the extra 90sec in the Debug flavor.
Btw is it expected that the NativeOnlyRelease flavor of this tests takes only 12sec, so much less?  

Re. 2 We've wanted to align this for a while, so you could look into this.

Reply:
@rectified95 please let's sync offline. thanks!

Reply:
Resolving as per offline discussion: the actual timing of the tests is definitely acceptable as it is. Should the test expand to more tasks (and time taken), it'd be worth reconsidering its displacement.